src/services/codefixes/inferFromUsage.ts

```diff
@@ -191,8 +200,9 @@ namespace ts.codefix {
             sourceFile,
             token.getStart(sourceFile));
 
-        Debug.assert(!!references, "Found no references!");
-        Debug.assert(references.length === 1, "Found more references than expected");
+        if (!references || references.length !== 1) {

```

Comment:
Is the `references.length !== 1` required? Seems weird to expect an array of exactly 1 object. If there are 2, is it better to return an empty array than ignore the 2nd unexpected element? Is it guaranteed/expected to always return 1 element in this context?

Reply:
This variable is probably badly named. `references` is not an array of references to a symbol, it's (effectively) an array of arrays; each element of the outer array refers to a different symbol (as could occur if you had a type and a value with the same name used in a position where either meaning might be valid, such as an `import` statement), and each subelement is the references to that symbol. In this case it would be very unusual (IOW we have no idea what's going on or what the right thing to do is) if we found *more than one symbol* matching a certain property.

Reply:
Is it better to remove the asserts or leave them? There is minimal user impact either way (though throwing errors probably isn't good for perf), since we merely end up swallowing the error and writing it as part of the failure response.